src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -3,12 +3,42 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import core.basesyntax.service.exceptions.PasswordException;
+import core.basesyntax.service.exceptions.UserExistException;
+import core.basesyntax.service.exceptions.UserNullException;
+import core.basesyntax.service.exceptions.UserYoungException;
 
 public class RegistrationServiceImpl implements RegistrationService {
+
+    private static final int MIN_USER_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        userValidate(user);
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new UserExistException("This user with login already exists");
+        }
+        return storageDao.add(user);
+    }
+
+    private void userValidate(User user) {
+        if (user == null) {
+            throw new RuntimeException("null is not a valid value");
```

Comment:
you have your own exception. Use it
```suggestion
            throw new UserRegistrationException("User cannot be null");
```